### PR TITLE
Fix subbatch test

### PR DESCRIPTION
**Pull request by Difers**
### PR Category

User Experience

### PR Types

Bug fixes

### Description

- vmm接口变化，相关单测修改

### 是否引起精度变化

否

### CODE REVIEW

File: tests/single_card_tests/test_moe_auto_subbatch.py [hunk 1]
```diff
@@ -475,24 +475,32 @@ def test_auto_subbatch_vs_ref(self):
...
 
         # test corner cases
         self.assertEqual(find_max_concurrent_subbatch_size([]), 1)
         self.assertEqual(find_max_sequence_subbatch_size(1024, 1), 1)
 
+        if old_large is not None:
+            MemoryAnalysisTool.vmm_large_all_block_info = old_large
+        else:
+            del MemoryAnalysisTool.vmm_large_all_block_info
         MemoryAnalysisTool.vmm_all_block_info = old_func
```
Comment (risemeup1111) [inline tests/single_card_tests/test_moe_auto_subbatch.py:504]: P2 **优先级：P2**
**问题：** 这里无条件创建 `MemoryAnalysisTool.vmm_large_all_block_info`，在旧 Paddle 运行时会让测试不再走 `vmm_free_and_growable_block_info()` 的 `AttributeError -> vmm_all_block_info()` fallback；同时恢复逻辑没有放进 `finally`，一旦中间断言或被测函数抛错，fake 的全局方法会留给后续测试。
**影响：** `vmm_utils.py` 仍显式兼容新旧两个 VMM 接口，这个测试改动会降低旧接口覆盖，并可能把后续失败变成污染后的连锁失败。
**处理要求：请针对该评论进行回复（同意并已修改请回复 Done，不同意请说明理由）。**
**建议：** 只在原本存在 `vmm_large_all_block_info` 时 patch 它，并用 `try/finally` 恢复全局状态：

```suggestion
        missing = object()
        old_large = getattr(
            MemoryAnalysisTool, "vmm_large_all_block_info", missing
        )
        old_func = MemoryAnalysisTool.vmm_all_block_info

        def set_blocks(blocks):
            if old_large is not missing:
                MemoryAnalysisTool.vmm_large_all_block_info = lambda: blocks
            MemoryAnalysisTool.vmm_all_block_info = lambda: blocks

        try:
            # test empty heap
            set_blocks([[]])
            info = vmm_free_and_growable_block_info()

            # test heap with separate free blocks
            set_blocks([[(1024, 0, True), (1024, 1024, False)]])
            info = vmm_free_and_growable_block_info()
            self.assertEqual(len(info), 2)

            # test corner cases
            self.assertEqual(find_max_concurrent_subbatch_size([]), 1)
            self.assertEqual(find_max_sequence_subbatch_size(1024, 1), 1)
        finally:
            if old_large is not missing:
                MemoryAnalysisTool.vmm_large_all_block_info = old_large
            elif hasattr(MemoryAnalysisTool, "vmm_large_all_block_info"):
                del MemoryAnalysisTool.vmm_large_all_block_info
            MemoryAnalysisTool.vmm_all_block_info = old_func
```

**Review by risemeup1111 — COMMENTED** [review]
已完成首轮审查，未发现需要阻塞合入的问题。测试里的全局 monkeypatch 恢复方式有一处非阻塞的隔离性建议，细节已放在行级评论中；当前 CI 还有部分任务在运行中，合入前仍需以最终 CI 结果为准。

<div align="right">
   <sup>Powered by Nyanpasu with gpt-5.5 xhigh, please check the suggestions carefully.</sup>
</div>

**Review by swgu98 — APPROVED** [review]
